<?xml version="1.0" encoding="utf-8"?>
<launch>
   <arg default="r1_sim" name="ugv_name"/>
   <arg default="basic" name="world_name"/>
   <arg default="true" name="enable_logging"/>
   <arg default="true" name="enable_ground_truth"/>
   <arg default="$(arg ugv_name)" name="log_file"/>
   <env name="GAZEBO_MODEL_PATH"
        value="${GAZEBO_MODEL_PATH}:$(find rotors_gazebo)/models"/>
   <env name="GAZEBO_RESOURCE_PATH"
        value="${GAZEBO_RESOURCE_PATH}:$(find rotors_gazebo)/models"/>
   <include file="$(find gazebo_ros)/launch/empty_world.launch">
      <arg name="world_name" value="$(find rcomv_uav)/worlds/$(arg world_name).world"/>
      <arg name="paused" value="true"/>
   </include>
   <arg default="true" name="wmsr_node"/>
   <arg default="5" name="n"/>
   <arg default="1" name="k"/>
   <arg default="0" name="F"/>
   <arg default="3" name="demo"/>
   <arg default="15" name="lead_x"/>
   <arg default="-15" name="lead_y"/>
   <arg default="0.4" name="Kp1"/>
   <arg default="0" name="Ki1"/>
   <arg default="0.1" name="Kd1"/>
   <arg default="4" name="Kp2"/>
   <arg default="0" name="Ki2"/>
   <arg default="1.5" name="Kd2"/>
   <!--upper level switch node-->
   <node launch-prefix="xterm -e" name="switch_node" output="screen" pkg="rcomv_r1"
         type="switch"/>
  <!-- Node which updates states of all agents -->
   <node name="builder_node" type="builder" pkg="state_graph_builder">
           <param name="n" value="$(arg n)"/>
         </node>
   <include file="$(find rcomv_r1)/launch/ugv_with_control.launch"><!--start of ugv1--><arg name="wmsr_node" value="$(arg wmsr_node)"/>
      <arg name="n" value="$(arg n)"/>
      <arg name="k" value="$(arg k)"/>
      <arg name="F" value="$(arg F)"/>
      <arg name="ugv_name" value="$(arg ugv_name)"/>
      <arg name="enable_logging" value="$(arg enable_logging)"/>
      <arg name="name_space" value="ugv1"/>
      <arg name="idx" value="1"/>
      <arg name="role" value="2"/>
      <arg name="x" value="10"/>
      <arg name="y" value="0.0"/>
      <arg name="lead_x" value="$(arg lead_x)"/>
      <arg name="lead_y" value="$(arg lead_y)"/>
      <arg name="demo" value="$(arg demo)"/>
      <arg name="Kp1" value="$(arg Kp1)"/>
      <arg name="Ki1" value="$(arg Ki1)"/>
      <arg name="Kd1" value="$(arg Kd1)"/>
      <arg name="Kp2" value="$(arg Kp2)"/>
      <arg name="Ki2" value="$(arg Ki2)"/>
      <arg name="Kd2" value="$(arg Kd2)"/>
   </include>
   <!--end of ugv1--><include file="$(find rcomv_r1)/launch/ugv_with_control.launch"><!--start of ugv2--><arg name="wmsr_node" value="$(arg wmsr_node)"/>
      <arg name="n" value="$(arg n)"/>
      <arg name="k" value="$(arg k)"/>
      <arg name="F" value="$(arg F)"/>
      <arg name="ugv_name" value="$(arg ugv_name)"/>
      <arg name="enable_logging" value="$(arg enable_logging)"/>
      <arg name="name_space" value="ugv2"/>
      <arg name="idx" value="2"/>
      <arg name="role" value="2"/>
      <arg name="x" value="3.0902"/>
      <arg name="y" value="9.5106"/>
      <arg name="lead_x" value="$(arg lead_x)"/>
      <arg name="lead_y" value="$(arg lead_y)"/>
      <arg name="demo" value="$(arg demo)"/>
      <arg name="Kp1" value="$(arg Kp1)"/>
      <arg name="Ki1" value="$(arg Ki1)"/>
      <arg name="Kd1" value="$(arg Kd1)"/>
      <arg name="Kp2" value="$(arg Kp2)"/>
      <arg name="Ki2" value="$(arg Ki2)"/>
      <arg name="Kd2" value="$(arg Kd2)"/>
   </include>
   <!--end of ugv2--><include file="$(find rcomv_r1)/launch/ugv_with_control.launch"><!--start of ugv3--><arg name="wmsr_node" value="$(arg wmsr_node)"/>
      <arg name="n" value="$(arg n)"/>
      <arg name="k" value="$(arg k)"/>
      <arg name="F" value="$(arg F)"/>
      <arg name="ugv_name" value="$(arg ugv_name)"/>
      <arg name="enable_logging" value="$(arg enable_logging)"/>
      <arg name="name_space" value="ugv3"/>
      <arg name="idx" value="3"/>
      <arg name="role" value="2"/>
      <arg name="x" value="-8.0902"/>
      <arg name="y" value="5.8779"/>
      <arg name="lead_x" value="$(arg lead_x)"/>
      <arg name="lead_y" value="$(arg lead_y)"/>
      <arg name="demo" value="$(arg demo)"/>
      <arg name="Kp1" value="$(arg Kp1)"/>
      <arg name="Ki1" value="$(arg Ki1)"/>
      <arg name="Kd1" value="$(arg Kd1)"/>
      <arg name="Kp2" value="$(arg Kp2)"/>
      <arg name="Ki2" value="$(arg Ki2)"/>
      <arg name="Kd2" value="$(arg Kd2)"/>
   </include>
   <!--end of ugv3--><include file="$(find rcomv_r1)/launch/ugv_with_control.launch"><!--start of ugv4--><arg name="wmsr_node" value="$(arg wmsr_node)"/>
      <arg name="n" value="$(arg n)"/>
      <arg name="k" value="$(arg k)"/>
      <arg name="F" value="$(arg F)"/>
      <arg name="ugv_name" value="$(arg ugv_name)"/>
      <arg name="enable_logging" value="$(arg enable_logging)"/>
      <arg name="name_space" value="ugv4"/>
      <arg name="idx" value="4"/>
      <arg name="role" value="2"/>
      <arg name="x" value="-8.0902"/>
      <arg name="y" value="-5.8779"/>
      <arg name="lead_x" value="$(arg lead_x)"/>
      <arg name="lead_y" value="$(arg lead_y)"/>
      <arg name="demo" value="$(arg demo)"/>
      <arg name="Kp1" value="$(arg Kp1)"/>
      <arg name="Ki1" value="$(arg Ki1)"/>
      <arg name="Kd1" value="$(arg Kd1)"/>
      <arg name="Kp2" value="$(arg Kp2)"/>
      <arg name="Ki2" value="$(arg Ki2)"/>
      <arg name="Kd2" value="$(arg Kd2)"/>
   </include>
   <!--end of ugv4--><include file="$(find rcomv_r1)/launch/ugv_with_control.launch"><!--start of ugv5--><arg name="wmsr_node" value="$(arg wmsr_node)"/>
      <arg name="n" value="$(arg n)"/>
      <arg name="k" value="$(arg k)"/>
      <arg name="F" value="$(arg F)"/>
      <arg name="ugv_name" value="$(arg ugv_name)"/>
      <arg name="enable_logging" value="$(arg enable_logging)"/>
      <arg name="name_space" value="ugv5"/>
      <arg name="idx" value="5"/>
      <arg name="role" value="2"/>
      <arg name="x" value="3.0902"/>
      <arg name="y" value="-50.5106"/>
      <arg name="lead_x" value="$(arg lead_x)"/>
      <arg name="lead_y" value="$(arg lead_y)"/>
      <arg name="demo" value="$(arg demo)"/>
      <arg name="Kp1" value="$(arg Kp1)"/>
      <arg name="Ki1" value="$(arg Ki1)"/>
      <arg name="Kd1" value="$(arg Kd1)"/>
      <arg name="Kp2" value="$(arg Kp2)"/>
      <arg name="Ki2" value="$(arg Ki2)"/>
      <arg name="Kd2" value="$(arg Kd2)"/>
   </include>
   <!--end of ugv5--></launch>
